agent/ui/pterm.go

```diff
@@ -0,0 +1,192 @@
+package ui
```

Comment:
Question, why not reusing what we have in `cli/ui`?

Reply:
Good question! I'm not using `cli/ui` to avoid a dependency of `cli -> agent`.  I'll unify them in the next PR, moving to a shared package.